### ISSUE TITLE
Cabby CLI push command fails when called for PDF file

### ISSUE THREAD

**Issue by traut** [eclecticiq/cabby#61]
Cabby can not push PDF files because it expects [non-binary file as input](https://github.com/eclecticiq/cabby/blob/master/cabby/cli/push.py#L41).

**Comment by arcsector** [comment]
I feel like this should be intended... I don't think that binary files fall under the "Automated eXchange" part of TAXII... I could be wrong but nowhere in the TAXII specs did I see a need for built-in PDF support. If there is a need for PDF support then there is a need for easy extraction of all good data in that PDF, and that sounds like a waste of resources, and completely antithetical to the "Structured" part of STIX. Why would anyone go through the extra step of PDF'ing information that was already written as XML or that was already going to be written as text anyways?

Just my two cents on the topic.

**Comment by traut** [comment]
@arcsector it's not particularly about PDF support, it is about binary files overall. If I'm not mistaken, even XML should be treated as a binary file.

Plus, I think TAXII 1.x is content agnostic, so it would be nice to fully support that

**Comment by arcsector** [comment]
That might be the kicker... TAXII 1.x may be content agnostic, but STIX necessitates a non-binary file...

I think this may only need to be handled on the input side of cabby... If python is type-agnostic, the user needs to be the one to deal with incoming content, and there shouldn't need to be any changes made, unless there's something I'm not thinking about.

This means that the only thing that needs to be answered is the push content, like your title says.

It's weird because it accepts a binary file in [`cabby.client11.push`](https://github.com/eclecticiq/cabby/blob/master/cabby/client11.py#L319), but then when i try to push it to test.taxiistand.com it throws a broken pipe SSL error! Code:
```python
from cabby import create_client

c = create_client()
cont = open('photo.jpg', 'rb').read()
#cont = 'f'
r = c.push(uri="https://test.taxiistand.com/read-write/services/inbox-all", content=cont, collection_names=["any-data"], content_binding="urn:stix.mitre.org:xml:1.1.1")
print(r)
```

Stack trace:
```shell
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/http/client.py", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1055, in _send_output
    self.send(chunk)
  File "/usr/local/lib/python3.7/http/client.py", line 977, in send
    self.sock.sendall(data)
  File "/usr/local/lib/python3.7/ssl.py", line 1012, in sendall
    v = self.send(byte_view[count:])
  File "/usr/local/lib/python3.7/ssl.py", line 981, in send
    return self._sslobj.write(data)
BrokenPipeError: [Errno 32] Broken pipe

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 445, in send
    timeout=timeout
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 638, in urlopen
    _stacktrace=sys.exc_info()[2])
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/retry.py", line 367, in increment
    raise six.reraise(type(error), error, _stacktrace)
  File "/usr/local/lib/python3.7/site-packages/urllib3/packages/six.py", line 685, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 354, in _make_request
    conn.request(method, url, **httplib_request_kw)
  File "/usr/local/lib/python3.7/http/client.py", line 1229, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1275, in _send_request
    self.endheaders(body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1224, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/local/lib/python3.7/http/client.py", line 1055, in _send_output
    self.send(chunk)
  File "/usr/local/lib/python3.7/http/client.py", line 977, in send
    self.sock.sendall(data)
  File "/usr/local/lib/python3.7/ssl.py", line 1012, in sendall
    v = self.send(byte_view[count:])
  File "/usr/local/lib/python3.7/ssl.py", line 981, in send
    return self._sslobj.write(data)
urllib3.exceptions.ProtocolError: ('Connection aborted.', BrokenPipeError(32, 'Broken pipe'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "testcabby.py", line 6, in <module>
    r = c.push(uri="https://test.taxiistand.com/read-write/services/inbox-all", content=cont, collection_names=["any-data"], content_binding="urn:stix.mitre.org:xml:1.1.1")
  File "/usr/local/lib/python3.7/site-packages/cabby/client11.py", line 332, in push
    service_type=const.SVC_INBOX)
  File "/usr/local/lib/python3.7/site-packages/cabby/abstract.py", line 190, in _execute_request
    timeout=self.timeout)
  File "/usr/local/lib/python3.7/site-packages/cabby/dispatcher.py", line 54, in send_taxii_request
    stream, headers = request_stream(session, url, request_body, timeout)
  File "/usr/local/lib/python3.7/site-packages/cabby/dispatcher.py", line 89, in request_stream
    headers=headers)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 559, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 512, in request
    resp = self.send(prep, **send_kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/sessions.py", line 622, in send
    r = adapter.send(request, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/requests/adapters.py", line 495, in send
    raise ConnectionError(err, request=request)
requests.exceptions.ConnectionError: ('Connection aborted.', BrokenPipeError(32, 'Broken pipe'))
```

**Comment by erwin-eiq** [comment]
Running the example from https://github.com/eclecticiq/cabby/issues/61#issuecomment-529079379 on current master results in `None` being printed, which according to https://github.com/eclecticiq/cabby/blob/ee5d46406f4cda82d9cd04ad61e6762349fe308a/cabby/dispatcher.py#L65 indicates success. I've verified this using pdb, the code indeed takes the success path.

Closing ticket, since it's apparently already fixed.

Thanks for reporting the issue @traut and for the reproducing testcase @arcsector .
